pyroclast/common/visualizable.py

```diff
@@ -98,3 +98,25 @@ def sensitivity_map(self, x, softmax=False):
                 result = tf.nn.softmax(result)
             grads = tape.gradient(result, x)
         return grads
+
+    def smooth_grad(self, x, n=50, sigma=0.01):
+        """Computes the smooth grad saliency-map
+
+        SmoothGrad works by averaging the gradient of n input images
+        that have been perturbed with Gausssian noise. Gaussian noise
+        is parameterized by the standard deviation (sigma).
+
+        Args:
+           x (np.array): shape [batch_size, ...data_shape]
```

Comment:
What is `x` though? I assume it should be some gradients, but it's never mentioned here

Reply:
x is a batch of input data. The method is constructed so that we can create the SmoothGrad maps for all inputs at once.

Reply:
I think that should be added to the docs before a merge. If you would.

Reply:
If you're just taking in input data, then the function as written is simply calculating an empirical average of noised data centered at the input. There isn't actually a gradient calculation in there anywhere.

Reply:
I didn't even do the correct implementation this the first time.